common/src/main/java/com/microsoft/identity/common/internal/util/AndroidKeyStoreUtil.java

```diff
@@ -447,11 +449,27 @@ public static synchronized SecretKey unwrap(@NonNull final byte[] wrappedKeyBlob
                 exception
         );
         if (exception instanceof InvalidKeyException) {
+            final android.security.KeyStoreException keyStoreException = findKeyStoreException(exception);
+            String ksMessage;
+            final String ksNumericErrorCode;
+            if (keyStoreException != null) {
+                ksMessage = keyStoreException.getMessage();
+                if (ksMessage == null) {
+                    ksMessage = "Keystore exception found, no error message";
+                }
+                ksNumericErrorCode = getNumericErrorCodeFromKeyStoreException(keyStoreException);
+            } else {
+                ksMessage = "No keystore exception found";
+                ksNumericErrorCode = "";
+            }
+
             final Attributes attributes = Attributes.builder()
                     .put(AttributeName.keystore_operation.name(), "unwrap")
                     .put(AttributeName.error_code.name(), errCode)
                     .put(AttributeName.error_type.name(), clientException.getClass().getSimpleName())
                     .put(AttributeName.keystore_exception_stack_trace.name(), ThrowableUtil.getStackTraceAsString(clientException))
+                    .put(AttributeName.keystore_exception_message.name(), ksMessage)
+                    .put(AttributeName.keystore_internal_error_code.name(), ksNumericErrorCode)
```

Comment:
>keystore_internal_error_code [](http://example.com/codeflow?start=39&length=28)

rename this because numeric errorcode is different from internal. internal would part of ksMessage.